src/spikeinterface/generation/hybrid_tools.py

```diff
@@ -0,0 +1,518 @@
+from __future__ import annotations
+
+import warnings
+from typing import Literal
+import numpy as np
+
+from ..core.template import Templates
+
+from ..core.generate import (
+    generate_templates,
+    generate_unit_locations,
+    generate_sorting,
+    InjectTemplatesRecording,
+    _ensure_seed,
+)
+from ..core.template_tools import get_template_extremum_channel
+from ..core.job_tools import split_job_kwargs
+from ..postprocessing.unit_localization import compute_monopolar_triangulation
+
+from .drift_tools import (
+    InjectDriftingTemplatesRecording,
+    DriftingTemplates,
+    make_linear_displacement,
+    interpolate_templates,
+    move_dense_templates,
+)
+
+
+def estimate_templates_from_recording(
+    recording, ms_before=2, ms_after=2, sorter_name="spykingcircus2", **run_sorter_kwargs
+):
+    """
+    Get templates from a recording. Internally, SpyKING CIRCUS 2 is used by default
+    with the only twist that the template matching step is not launched. Instead, a Template
+    object is returned based on the results of the clustering. Other sorters can be invoked
+    with the `sorter_name` and `run_sorter_kwargs` parameters.
+
+    Parameters
+    ----------
+    ms_before : float
+        The time before peaks of templates
+    ms_after : float
+        The time after peaks of templates
+    sorter_name : str
+        The sorter to be used in order to get some fast clustering
+    run_sorter_kwargs : dict
+        The parameters to provide to the run_sorter function of spikeinterface
+
+
+    sorter_params: keyword arguments for `spyking_circus2` function
+
+    Returns
+    -------
+    templates: Templates
+        The found templates
+    """
+    from spikeinterface.sorters.runsorter import run_sorter
+    from spikeinterface.core.template import Templates
+    from spikeinterface.core.waveform_tools import estimate_templates
+
+    if sorter_name == "spykingcircus2":
+        if "matching" not in run_sorter_kwargs:
+            run_sorter_kwargs["matching"] = {"method": None}
```

Comment:
Why having sorter specifics here ?

Reply:
Because we don't need to launch the template macthing step to get the Templates. If some sorters can have the option to get templates without a proper sorting, then we should use it

Reply:
not in this PR, but we could make a `hybrid_sorter` which is designed for this:
- high defautk peak detection threshold
- conservative clustering
- no TM

Reply:
so we don't need to make custom adjustments to existing end-to-end sorters